diff --git a/src/MQTTAsync.c b/src/MQTTAsync.c
index b866250..8c4d995 100644
--- a/src/MQTTAsync.c
+++ b/src/MQTTAsync.c
@@ -1087,7 +1087,7 @@ static int MQTTAsync_addCommand(MQTTAsync_queuedCommand* command, int command_si
 	{
 		ListAppend(commands, command, command_size);
 #if !defined(NO_PERSISTENCE)
-		if (command->client->c->persistence)
+        if (command->client->c->persistence && (command->command.type != PUBLISH || command->command.details.pub.qos > 0))
 			MQTTAsync_persistCommand(command);
 #endif
 	}
